src/identify/index.js

```diff
@@ -314,26 +314,40 @@ class IdentifyService {
    * @return {Uint8Array}
    */
   async _getSelfPeerRecord () {
-    const selfSignedPeerRecord = this.peerStore.addressBook.getRawEnvelope(this.peerId)
+    // Update self peer record if needed
+    await this._createOrUpdateSelfPeerRecord()
 
-    // TODO: support invalidation when dynamic multiaddrs are supported
-    if (selfSignedPeerRecord) {
-      return selfSignedPeerRecord
+    return this.peerStore.addressBook.getRawEnvelope(this.peerId)
+  }
+
+  /**
+   * Creates or updates the self peer record if it exists and is outdated.
+   * @return {Promise<void>}
+   */
+  async _createOrUpdateSelfPeerRecord () {
```

Comment:
This method addition seems unnecessary? This just makes `_getSelfPeerRecord` pretty much useless.

Also, we should not be doing update checks in here, we should just update the record when an actual change happens. This is going to cause a lot of unnecessary address checks.